### PR TITLE
feat(withdraw pledge): withdraw pledge verify

### PR DESCRIPTION
**Pull request by JacobDenver007**

### CODE REVIEW

File: contracts/toCKB-typescript/src/switch/withdraw_pledge/mod.rs [hunk 1]
```diff
@@ -1,7 +1,39 @@
...
+use crate::utils::config::{PLEDGE, SINCE_N1};
+use crate::utils::types::{Error, ToCKBCellDataView};
+use ckb_std::ckb_constants::Source;
+use ckb_std::ckb_types::prelude::*;
+use ckb_std::high_level::{load_cell, load_input_since, QueryIter};
 use core::result::Result;
 
-pub fn verify(_toCKB_data_tuple: &ToCKBCellDataTuple) -> Result<(), Error> {
+pub fn verify_since() -> Result<(), Error> {
+    let since = load_input_since(0, Source::Input)?;
```
Comment (mkxbl) [inline contracts/toCKB-typescript/src/switch/withdraw_pledge/mod.rs:10]: should be `Source::GroupInput`

File: contracts/toCKB-typescript/src/switch/withdraw_pledge/mod.rs [hunk 2]
```diff
@@ -1,7 +1,39 @@
...
+    if since != SINCE_N1 {
+        return Err(Error::InvariantDataMutated);
```
Comment (mkxbl) [inline contracts/toCKB-typescript/src/switch/withdraw_pledge/mod.rs:12]: use `Error::InputSinceInvalid`

File: contracts/toCKB-typescript/src/switch/withdraw_pledge/mod.rs [hunk 3]
```diff
@@ -1,7 +1,39 @@
...
+    }
     Ok(())
 }
+
+pub fn verify_capacity(input_toCKB_data: &ToCKBCellDataView) -> Result<(), Error> {
+    let outputs = QueryIter::new(load_cell, Source::GroupOutput);
```
Comment (mkxbl) [inline contracts/toCKB-typescript/src/switch/withdraw_pledge/mod.rs:18]: should be `Source::Output`

File: contracts/toCKB-typescript/src/utils/config.rs [hunk 4]
```diff
@@ -1 +1,2 @@
 pub const PLEDGE: u64 = 10000;
+pub const SINCE_N1: u64 = 0;
```
Comment (mkxbl) [inline contracts/toCKB-typescript/src/utils/config.rs:2]: why `SINCE_N1 = 0`

File: contracts/toCKB-typescript/src/switch/withdraw_pledge/mod.rs [hunk 5]
```diff
@@ -1,7 +1,39 @@
...
+fn verify_since() -> Result<(), Error> {
+    let since = load_input_since(0, Source::GroupInput)?;
+    if since != SINCE_WITHDRAW_PLEDGE {
+        return Err(Error::InputSinceInvalid);
+    }
     Ok(())
 }
+
+fn verify_capacity(input_toCKB_data: &ToCKBCellDataView) -> Result<(), Error> {
+    let outputs = QueryIter::new(load_cell, Source::Output);
```
Comment (mkxbl) [inline contracts/toCKB-typescript/src/switch/withdraw_pledge/mod.rs:18]:
```suggestion
    let outputs = QueryIter::new(load_cell_lock, Source::Output);
```

File: contracts/toCKB-typescript/src/switch/withdraw_pledge/mod.rs [hunk 6]
```diff
@@ -1,7 +1,39 @@
...
+    let mut sum = 0;
+    for output in outputs {
+        if output.lock().as_bytes().as_ref() == input_toCKB_data.user_lockscript.as_ref() {
+            sum += output.capacity().unpack()
+        }
+    }
+
+    if sum != PLEDGE {
```
Comment (mkxbl) [inline contracts/toCKB-typescript/src/switch/withdraw_pledge/mod.rs:26]:
```suggestion
    if sum < PLEDGE {
```

File: contracts/toCKB-typescript/src/switch/withdraw_pledge/mod.rs (hunk 5 above)
Comment (JacobDenver007) [reply contracts/toCKB-typescript/src/switch/withdraw_pledge/mod.rs:19]: is ```load_cell_lock``` equal to cell.lock().as_bytes()?

Comment (mkxbl) [reply contracts/toCKB-typescript/src/switch/withdraw_pledge/mod.rs:19]: yep, but with less cycle-cost

Comment (mkxbl) [reply contracts/toCKB-typescript/src/switch/withdraw_pledge/mod.rs:19]: yep, but with less cycle-cost

**Pull request merged**